cmd/soroban-rpc/internal/preflight/preflight.go

```diff
@@ -36,7 +35,7 @@ type snapshotSourceHandle struct {
 }
 
 const (
-	defaultInstructionLeeway uint64 = 3000000
+	defaultInstructionLeeway uint64 = 0
```

Comment:
@dmkozh This will surely have an impact for the end user.

Are you sure this won't lead to the instruction/fee estimation falling short for the end user? I think it's better to overestimate.

Reply:
Tightening the simulation estimates is the main functional goal of this whole project and this change reflects that (+3M leeway has always been a hack and we wanted to remove it ASAP). As you can see, the integration tests are still passing and there is also env side coverage for all the host fns and pretty complex setups with auth and cross-contract calls. These tests are passing with just 1% leeway.

I also think it's better to store defaults in the simulation library itself (https://github.com/stellar/rs-soroban-env/blob/8c9ab83c406bd86f56d52eae3e39dccf6b45b3da/soroban-simulation/src/simulation.rs#L315). This way they can reflect the state of simulation accuracy. That said, it is easy to modify `SimulationAdjustmentConfig` on the RPC side in case if we find that something is off without doing any changes in the simulation crate itself. The same goes for providing user-customizable leeways for more resources - it should be easy to do that without changing anything about the simulation library.

Reply:
Alright, makes sense!